static/js/tools/shared/info_examples.tsx

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * A component which renders a memoizable set of links configured and stored in the page window object.
+ */
+
+import React from "react";
+
+// Type for a single link.
+interface InfoLink {
+  text: string;
+  url: string;
+}
+
+// Type for a single row or bullet (which could contain nested rows of related examples).
+interface InfoRow {
+  header: string;
+  // If preposition is defined, then examples are expected.
+  preposition?: string;
+  examples?: InfoLink[];
+  // If preposition is not defined, then subheaders are expected and will
+  // be used to generate links similar to timelines.
+  subheaders: InfoSubheader[];
+}
+
+interface InfoSubheader {
+  title: string;
+  examples: InfoLink[];
+}
+
+declare global {
```

Comment:
I wonder if there are other alternatives to the global Window solution, but this feels like a concise solution in this scenario. Please verify switching between the pages work as intended.

Reply:
that's nice!